--- a/modules/gui/qt4/actions_manager.cpp	2016-07-29 00:35:50.000000000 +0200
+++ b/modules/gui/qt4/actions_manager.cpp	2018-07-28 17:11:25.113847366 +0200
@@ -26,6 +26,7 @@
 #endif
 
 #include "qt4.hpp"
+#include <QAction>
 
 #include <vlc_vout.h>
 #include <vlc_keys.h>
--- a/modules/gui/qt4/components/simple_preferences.cpp	2016-12-29 11:38:11.000000000 +0100
+++ b/modules/gui/qt4/components/simple_preferences.cpp	2018-07-28 17:13:34.791700989 +0200
@@ -39,6 +39,7 @@
 #include <QSignalMapper>
 #include <QVBoxLayout>
 #include <QScrollArea>
+#include <QButtonGroup>
 
 #include <QStyleFactory>
 #include <QSettings>
