commit c8a7d1276bbda6e8102a44fb3f5473fa9374dad3
Author: Holger Hoffstätte <holger@applied-asynchrony.com>
Date:   Wed Mar 3 17:00:14 2021 +0100

    Silence unnecessary nocow warning

diff --git a/src/swaps.c b/src/swaps.c
index e477344..03ba91a 100644
--- a/src/swaps.c
+++ b/src/swaps.c
@@ -459,7 +459,6 @@ int set_no_cow(int fd)
     if (errno == ENOTSUP)
     {
       nocow_ok = false;
-      logm(LOG_NOTICE, "NOCOW attribute not supported!");
     }
     else return err;
   }
