Disable -Werror for endian tests.

--- a/libs/endian/test/Jamfile.v2
+++ b/libs/endian/test/Jamfile.v2
@@ -22,7 +22,7 @@ project
     <toolset>gcc:<cxxflags>-Wno-long-long
     <toolset>gcc-4.4:<cxxflags>-Wno-strict-aliasing
     <toolset>gcc-4.4:<cxxflags>-Wno-sign-compare
-    <toolset>gcc:<warnings-as-errors>on
+    <toolset>gcc:<warnings-as-errors>off
 
     <toolset>clang:<cxxflags>-Wno-long-long
     <toolset>clang:<warnings-as-errors>on
