From dd482b53f107e9dcdf036684f1e66394af88c61d Mon Sep 17 00:00:00 2001
From: Dziban303 <dziban303@users.noreply.github.com>
Date: Tue, 24 Feb 2026 18:32:50 -0600
Subject: [PATCH] fix for nil report

---
 collector/sourcestats.go | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/collector/sourcestats.go b/collector/sourcestats.go
index ba3189b..8c91830 100644
--- a/collector/sourcestats.go
+++ b/collector/sourcestats.go
@@ -125,6 +125,10 @@ func (e Exporter) getSourcestatsMetrics(logger *slog.Logger, ch chan<- prometheu
 	}
 
 	for _, r := range results {
+		if r.IPAddr == nil {
+			logger.Debug("Skipping source with nil IP address")
+			continue
+		}
 		sourceAddress := r.IPAddr.String()
 		sourceName := e.dnsLookup(logger, r.IPAddr)
 
