From 70f000d92e425ecf24253120e56b2ef655c02714 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Lukas=20F=C3=BClling?= <lukas@k40s.net>
Date: Sat, 20 Jan 2018 13:21:33 +0100
Subject: [PATCH] add send/receive rates

---
 main.go | 24 ++++++++++++++++++++++++
 1 file changed, 24 insertions(+)

diff --git a/main.go b/main.go
index 0d3f641..9700f64 100644
--- a/main.go
+++ b/main.go
@@ -103,6 +103,30 @@ var metrics = []*Metric{
 		),
 		MetricType: prometheus.CounterValue,
 	},
+	{
+		Service: "urn:schemas-upnp-org:service:WANCommonInterfaceConfig:1",
+                Action:  "GetAddonInfos",
+                Result:  "ByteSendRate",
+                Desc: prometheus.NewDesc(
+                        "gateway_wan_bytes_send_rate",
+                        "byte send rate on gateway WAN interface",
+                        []string{"gateway"},
+                        nil,
+                ),
+                MetricType: prometheus.GaugeValue,
+        },
+        {
+		Service: "urn:schemas-upnp-org:service:WANCommonInterfaceConfig:1",
+                Action:  "GetAddonInfos",
+                Result:  "ByteReceiveRate",
+                Desc: prometheus.NewDesc(
+                        "gateway_wan_bytes_receive_rate",
+                        "byte receive rate on gateway WAN interface",
+                        []string{"gateway"},
+                        nil,
+                ),
+                MetricType: prometheus.GaugeValue,
+        },
 	{
 		Service: "urn:schemas-upnp-org:service:WANCommonInterfaceConfig:1",
 		Action:  "GetCommonLinkProperties",
