https://github.com/knorrie/python-btrfs/issues/55

--- a//dev/null
+++ b/setup.cfg
@@ -0,0 +1,2 @@
+[metadata]
+license = GNU Lesser General Public License v3 (LGPLv3)
--- a/setup.py
+++ b/setup.py
@@ -56,7 +56,6 @@ setup(
         'Development Status :: 5 - Production/Stable',
         'Intended Audience :: Developers',
         'Intended Audience :: System Administrators',
-        'License :: OSI Approved :: GNU Lesser General Public License v3 (LGPLv3)',
         'Programming Language :: Python :: 3 :: Only',
         'Programming Language :: Python :: Implementation :: CPython',
         'Operating System :: POSIX :: Linux',
