diff -rup mailfilter-0.8.5/src/rcfile.cc mailfilter-0.8.5.fix/src/rcfile.cc
--- mailfilter-0.8.5/src/rcfile.cc	2016-07-26 18:25:53.000000000 +0200
+++ mailfilter-0.8.5.fix/src/rcfile.cc	2016-07-28 17:08:14.016421311 +0200
@@ -1528,7 +1528,7 @@ ECHO;
  * This constructor simply maintains backward compatibility.
  * DEPRECATED
  */
-yyFlexLexer::yyFlexLexer( FLEX_STD istream* arg_yyin, FLEX_STD ostream* arg_yyout ):
+yyFlexLexer::yyFlexLexer( std::istream* arg_yyin, std::ostream* arg_yyout ):
 	yyin(arg_yyin ? arg_yyin->rdbuf() : std::cin.rdbuf()),
 	yyout(arg_yyout ? arg_yyout->rdbuf() : std::cout.rdbuf())
 {
diff -rup mailfilter-0.8.5/src/rfc822.cc mailfilter-0.8.5.fix/src/rfc822.cc
--- mailfilter-0.8.5/src/rfc822.cc	2016-07-26 18:25:53.000000000 +0200
+++ mailfilter-0.8.5.fix/src/rfc822.cc	2016-07-28 17:08:34.169913216 +0200
@@ -931,7 +931,7 @@ case YY_STATE_EOF(TAG_READY):
  * This constructor simply maintains backward compatibility.
  * DEPRECATED
  */
-yyFlexLexer::yyFlexLexer( FLEX_STD istream* arg_yyin, FLEX_STD ostream* arg_yyout ):
+yyFlexLexer::yyFlexLexer( std::istream* arg_yyin, std::ostream* arg_yyout ):
 	yyin(arg_yyin ? arg_yyin->rdbuf() : std::cin.rdbuf()),
 	yyout(arg_yyout ? arg_yyout->rdbuf() : std::cout.rdbuf())
 {
