diff -rup zstd-1.3.6/programs/zstd.1 zstd-1.3.6+/programs/zstd.1
--- zstd-1.3.6/programs/zstd.1	2018-10-05 18:34:52.000000000 +0200
+++ zstd-1.3.6+/programs/zstd.1	2018-10-05 21:16:54.297195897 +0200
@@ -116,7 +116,7 @@ Note: If \fBwindowLog\fR is set to large
 .
 .TP
 \fB\-T#\fR, \fB\-\-threads=#\fR
-Compress using \fB#\fR working threads (default: 1)\. If \fB#\fR is 0, attempt to detect and use the number of physical CPU cores\. In all cases, the nb of threads is capped to ZSTDMT_NBTHREADS_MAX==200\. This modifier does nothing if \fBzstd\fR is compiled without multithread support\.
+Compress using \fB#\fR working threads (default: 0)\. If \fB#\fR is 0, attempt to detect and use the number of physical CPU cores\. In all cases, the nb of threads is capped to ZSTDMT_NBTHREADS_MAX==200\. This modifier does nothing if \fBzstd\fR is compiled without multithread support\.
 .
 .TP
 \fB\-\-single\-thread\fR
diff -rup zstd-1.3.6/programs/zstdcli.c zstd-1.3.6+/programs/zstdcli.c
--- zstd-1.3.6/programs/zstdcli.c	2018-10-05 18:34:52.000000000 +0200
+++ zstd-1.3.6+/programs/zstdcli.c	2018-10-05 21:18:21.224568623 +0200
@@ -141,7 +141,7 @@ static int usage_advanced(const char* pr
     DISPLAY( "--fast[=#]: switch to ultra fast compression level (default: %u)\n", 1);
     DISPLAY( "--adapt : dynamically adapt compression level to I/O conditions \n");
 #ifdef ZSTD_MULTITHREAD
-    DISPLAY( " -T#    : spawns # compression threads (default: 1, 0==# cores) \n");
+    DISPLAY( " -T#    : spawns # compression threads (default: 0 (= # cores) \n");
     DISPLAY( " -B#    : select size of each job (default: 0==automatic) \n");
 #endif
     DISPLAY( "--no-dictID : don't write dictID into header (dictionary compression)\n");
@@ -527,7 +527,7 @@ int main(int argCount, const char* argv[
     g_displayOut = stderr;
     programName = lastNameFromPath(programName);
 #ifdef ZSTD_MULTITHREAD
-    nbWorkers = 1;
+    nbWorkers = 0;
 #endif
 
     /* preset behaviors */
