From f2839f4c5e8b7a182e1bc7bdadabbf2007b91fde Mon Sep 17 00:00:00 2001
From: Lefteris Chatzimparmpas <lefcha@fastmail.com>
Date: Sat, 18 Sep 2021 12:30:07 +0300
Subject: [PATCH] Don't destroy session on recover mechanism login failures.

---
 src/request.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/request.c b/src/request.c
index add7b39..62d6c9c 100644
--- a/src/request.c
+++ b/src/request.c
@@ -292,8 +292,10 @@ request_login(session **ssnptr, const char *server, const char *port, const
 abort:
 	close_connection(ssn);
 fail:
-	session_destroy(ssn);
-	ssn = NULL;
+	if (!*ssnptr) {
+		session_destroy(ssn);
+		ssn = NULL;
+	}
 
 	return -1;
 }
