From daef6161e4801bc0f8c6719d37c8392cc702e247 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Holger=20Hoffst=C3=A4tte?= <holger@applied-asynchrony.com>
Date: Sat, 2 Oct 2021 20:11:05 +0200
Subject: [PATCH] Add missing braces to silence warning and fix potential state management issues. (#236)

See #235
---
 src/request.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/request.c b/src/request.c
index 62d6c9c..0f4cd4a 100644
--- a/src/request.c
+++ b/src/request.c
@@ -54,9 +54,10 @@ int send_continuation(session *ssn, const char *data, size_t len);
 			if (request_login(&ssn, NULL, NULL, NULL, NULL,	       \
 			    NULL, NULL) != -1)				       \
 				return STATUS_NONE;			       \
-		} else							       \
+		} else {						       \
 			session_destroy(ssn);				       \
 			ssn = NULL;					       \
+		}							       \
 		return -1;						       \
 	}
 
