From 0b809d9726d1db17e17ffd87b6aeb4e079d82a37 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Holger=20Hoffst=C3=A4tte?= <holger@applied-asynchrony.com>
Date: Sat, 18 Aug 2018 19:09:09 +0200
Subject: [PATCH] Initial change to read /proc/diskstats with linux-4.19

---
 collector/diskstats_linux.go | 32 ++++++++++++++++++++++++++++++++
 1 file changed, 32 insertions(+)

diff --git a/collector/diskstats_linux.go b/collector/diskstats_linux.go
index c80db4c9..c3a45203 100644
--- a/collector/diskstats_linux.go
+++ b/collector/diskstats_linux.go
@@ -162,6 +162,38 @@ func NewDiskstatsCollector() (Collector, error) {
 				), valueType: prometheus.CounterValue,
 				factor: .001,
 			},
+			{
+				desc: prometheus.NewDesc(
+					prometheus.BuildFQName(namespace, diskSubsystem, "discards_completed_total"),
+					"The total number of discards completed successfully.",
+					diskLabelNames,
+					nil,
+				), valueType: prometheus.CounterValue,
+			},
+			{
+				desc: prometheus.NewDesc(
+					prometheus.BuildFQName(namespace, diskSubsystem, "discards_merged_total"),
+					"The total number of discards merged. See https://www.kernel.org/doc/Documentation/iostats.txt.",
+					diskLabelNames,
+					nil,
+				), valueType: prometheus.CounterValue,
+			},
+			{
+				desc: prometheus.NewDesc(
+					prometheus.BuildFQName(namespace, diskSubsystem, "discarded_sectors_total"),
+					"The total number of sectors discard successfully.",
+					diskLabelNames,
+					nil,
+				), valueType: prometheus.CounterValue,
+			},
+			{
+				desc: prometheus.NewDesc(
+					prometheus.BuildFQName(namespace, diskSubsystem, "discard_time_milliseconds_total"),
+					"This is the total number of milliseconds spent by all discards.",
+					diskLabelNames,
+					nil,
+				), valueType: prometheus.CounterValue,
+			},
 		},
 	}, nil
 }
