
Only do packet timeout scanning & health check scheduling every 10s.

--- dnsdist-1.7.1/dnsdist.cc~	2022-04-22 17:05:41.000000000 +0200
+++ dnsdist-1.7.1/dnsdist.cc	2022-04-26 00:48:11.586526407 +0200
@@ -1842,7 +1842,7 @@ static void healthChecksThread()
 {
   setThreadName("dnsdist/healthC");
 
-  static const int interval = 1;
+  static const int interval = 10;
 
   for(;;) {
     sleep(interval);
