From c42df46af74f9af4158eebb51bfba3d1d8fb1a42 Mon Sep 17 00:00:00 2001
From: Otto Moerbeek <otto.moerbeek@open-xchange.com>
Date: Tue, 19 Mar 2024 10:47:59 +0100
Subject: [PATCH] dnsdist: also handle EHOSTUNRERACH as a case for reconnecting the socket

---
 a/dnsdist.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dnsdist.cc b/dnsdist.cc
index a73cf87c993e..36fb6ba7731e 100644
--- a/dnsdist.cc
+++ b/dnsdist.cc
@@ -1255,7 +1255,7 @@ ssize_t udpClientSendRequestToBackend(const std::shared_ptr<DownstreamState>& ss
        because it's not using the same socket.
     */
     if (!healthCheck) {
-      if (savederrno == EINVAL || savederrno == ENODEV || savederrno == ENETUNREACH || savederrno == EBADF) {
+      if (savederrno == EINVAL || savederrno == ENODEV || savederrno == ENETUNREACH || savederrno == EHOSTUNREACH || savederrno == EBADF) {
         ss->reconnect();
       }
       ss->reportTimeoutOrError();
