From b0b3480b98d41db821f681183f45d5d08db02f93 Mon Sep 17 00:00:00 2001
From: Remi Gacogne <remi.gacogne@powerdns.com>
Date: Mon, 18 Mar 2024 16:16:17 +0100
Subject: [PATCH] dnsdist: Properly account the failure to forward a query to a backend

---
 dnsdist.cc | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/dnsdist.cc b/dnsdist.cc
index f836e1320000..06dc4781189d 100644
--- a/dnsdist.cc
+++ b/dnsdist.cc
@@ -1260,8 +1260,11 @@ ssize_t udpClientSendRequestToBackend(const std::shared_ptr<DownstreamState>& ss
        We don't want to reconnect the real socket if the healthcheck failed,
        because it's not using the same socket.
     */
-    if (!healthCheck && (savederrno == EINVAL || savederrno == ENODEV || savederrno == ENETUNREACH || savederrno == EBADF)) {
-      ss->reconnect();
+    if (!healthCheck) {
+      if (savederrno == EINVAL || savederrno == ENODEV || savederrno == ENETUNREACH || savederrno == EBADF) {
+        ss->reconnect();
+      }
+      ss->reportTimeoutOrError();
     }
   }
 
