From ac4afa03883d795313cf3f6896b53e50d3c363ce Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Holger=20Hoffst=C3=A4tte?= <holger@applied-asynchrony.com>
Date: Sat, 19 Jun 2021 10:27:58 +0200
Subject: [PATCH] dnsdist: Do not report metrics about down upstream servers

---
 dnsdist-web.cc | 42 ++++++++++++++++++++++--------------------
 1 file changed, 22 insertions(+), 20 deletions(-)

diff --git a/dnsdist-web.cc b/dnsdist-web.cc
index 9d31b3aefae..110ff4c54a6 100644
--- a/dnsdist-web.cc
+++ b/dnsdist-web.cc
@@ -504,26 +504,28 @@ static void handlePrometheus(const YaHTTP::Request& req, YaHTTP::Response& resp)
                                          % serverName % state->remote.toStringWithPort());
 
     output << statesbase << "status"                       << label << " " << (state->isUp() ? "1" : "0")        << "\n";
-    output << statesbase << "queries"                      << label << " " << state->queries.load()              << "\n";
-    output << statesbase << "responses"                    << label << " " << state->responses.load()            << "\n";
-    output << statesbase << "drops"                        << label << " " << state->reuseds.load()              << "\n";
-    output << statesbase << "latency"                      << label << " " << state->latencyUsec/1000.0          << "\n";
-    output << statesbase << "senderrors"                   << label << " " << state->sendErrors.load()           << "\n";
-    output << statesbase << "outstanding"                  << label << " " << state->outstanding.load()          << "\n";
-    output << statesbase << "order"                        << label << " " << state->order                       << "\n";
-    output << statesbase << "weight"                       << label << " " << state->weight                      << "\n";
-    output << statesbase << "tcpdiedsendingquery"          << label << " " << state->tcpDiedSendingQuery         << "\n";
-    output << statesbase << "tcpdiedreadingresponse"       << label << " " << state->tcpDiedReadingResponse      << "\n";
-    output << statesbase << "tcpgaveup"                    << label << " " << state->tcpGaveUp                   << "\n";
-    output << statesbase << "tcpreadtimeouts"              << label << " " << state->tcpReadTimeouts             << "\n";
-    output << statesbase << "tcpwritetimeouts"             << label << " " << state->tcpWriteTimeouts            << "\n";
-    output << statesbase << "tcpconnecttimeouts"           << label << " " << state->tcpConnectTimeouts         << "\n";
-    output << statesbase << "tcpcurrentconnections"        << label << " " << state->tcpCurrentConnections       << "\n";
-    output << statesbase << "tcpmaxconcurrentconnections"  << label << " " << state->tcpMaxConcurrentConnections << "\n";
-    output << statesbase << "tcpnewconnections"            << label << " " << state->tcpNewConnections           << "\n";
-    output << statesbase << "tcpreusedconnections"         << label << " " << state->tcpReusedConnections        << "\n";
-    output << statesbase << "tcpavgqueriesperconn"         << label << " " << state->tcpAvgQueriesPerConnection  << "\n";
-    output << statesbase << "tcpavgconnduration"           << label << " " << state->tcpAvgConnectionDuration    << "\n";
+    if (state->isUp()) {
+      output << statesbase << "queries"                      << label << " " << state->queries.load()              << "\n";
+      output << statesbase << "responses"                    << label << " " << state->responses.load()            << "\n";
+      output << statesbase << "drops"                        << label << " " << state->reuseds.load()              << "\n";
+      output << statesbase << "latency"                      << label << " " << state->latencyUsec/1000.0          << "\n";
+      output << statesbase << "senderrors"                   << label << " " << state->sendErrors.load()           << "\n";
+      output << statesbase << "outstanding"                  << label << " " << state->outstanding.load()          << "\n";
+      output << statesbase << "order"                        << label << " " << state->order                       << "\n";
+      output << statesbase << "weight"                       << label << " " << state->weight                      << "\n";
+      output << statesbase << "tcpdiedsendingquery"          << label << " " << state->tcpDiedSendingQuery         << "\n";
+      output << statesbase << "tcpdiedreadingresponse"       << label << " " << state->tcpDiedReadingResponse      << "\n";
+      output << statesbase << "tcpgaveup"                    << label << " " << state->tcpGaveUp                   << "\n";
+      output << statesbase << "tcpreadtimeouts"              << label << " " << state->tcpReadTimeouts             << "\n";
+      output << statesbase << "tcpwritetimeouts"             << label << " " << state->tcpWriteTimeouts            << "\n";
+      output << statesbase << "tcpconnecttimeouts"           << label << " " << state->tcpConnectTimeouts          << "\n";
+      output << statesbase << "tcpcurrentconnections"        << label << " " << state->tcpCurrentConnections       << "\n";
+      output << statesbase << "tcpmaxconcurrentconnections"  << label << " " << state->tcpMaxConcurrentConnections << "\n";
+      output << statesbase << "tcpnewconnections"            << label << " " << state->tcpNewConnections           << "\n";
+      output << statesbase << "tcpreusedconnections"         << label << " " << state->tcpReusedConnections        << "\n";
+      output << statesbase << "tcpavgqueriesperconn"         << label << " " << state->tcpAvgQueriesPerConnection  << "\n";
+      output << statesbase << "tcpavgconnduration"           << label << " " << state->tcpAvgConnectionDuration    << "\n";
+    }
   }
 
   const string frontsbase = "dnsdist_frontend_";
