From: Pieter Lexis <pieter.lexis@powerdns.com>
Date: Fri, 27 Nov 2020 13:23:05 +0100
Subject: [PATCH] dnsdist: Drop second argument in `SpoofAction`

Fixes #9783

[HH: backported to 1.5.x]

diff --git a/pdns/dnsdist-lua-actions.cc b/pdns/dnsdist-lua-actions.cc
index 9ea2eedefd..29f1f2045a 100644
--- a/dnsdist-lua-actions.cc
+++ b/dnsdist-lua-actions.cc
@@ -1604,7 +1604,7 @@ void setupLuaActions()
       return std::shared_ptr<DNSAction>(new QPSPoolAction(limit, a));
     });
 
-  g_lua.writeFunction("SpoofAction", [](boost::variant<std::string,vector<pair<int, std::string>>> inp, boost::optional<std::string> b, boost::optional<responseParams_t> vars) {
+  g_lua.writeFunction("SpoofAction", [](boost::variant<std::string,vector<pair<int, std::string>>> inp, boost::optional<responseParams_t> vars) {
       vector<ComboAddress> addrs;
       if(auto s = boost::get<std::string>(&inp))
         addrs.push_back(ComboAddress(*s));
@@ -1613,9 +1613,6 @@ void setupLuaActions(LuaContext& luaCtx)
         for(const auto& a: v)
           addrs.push_back(ComboAddress(a.second));
       }
-      if(b) {
-        addrs.push_back(ComboAddress(*b));
-      }
 
       auto ret = std::shared_ptr<DNSAction>(new SpoofAction(addrs));
       auto sa = std::dynamic_pointer_cast<SpoofAction>(ret);
