diff -rup java-config-2.2.0/src/java-config-2 java-config-2.2.0-hh/src/java-config-2
--- java-config-2.2.0/src/java-config-2	2013-02-14 23:12:09.000000000 +0100
+++ java-config-2.2.0-hh/src/java-config-2	2020-04-29 22:05:31.461021020 +0200
@@ -199,7 +199,7 @@ def set_user_vm(option, opt, value, pars
     if not vm:
         fatalError("Could not find a vm matching: %s" % value)
     else:
-        if os.getuid() is 0:
+        if os.getuid() == 0:
             fatalError("The user 'root' should always use the System VM")
         else:
             try:
@@ -227,7 +227,7 @@ def user_classpath_target():
 # Deprecated
 def set_system_classpath(option, opt, value, parser):
     deprecation_notice()
-    if os.getuid() is 0:
+    if os.getuid() == 0:
         pkgs = value.split(',')
         manager.set_classpath(system_classpath_target(), pkgs)
         
@@ -252,7 +252,7 @@ def set_user_classpath(option, opt, valu
 # Deprecated
 def append_system_classpath(option, opt, value, parser):
     deprecation_notice()
-    if os.getuid() is 0:
+    if os.getuid() == 0:
         pkgs = value.split(',')
         manager.append_classpath(system_classpath_target(), pkgs)
 
@@ -277,7 +277,7 @@ def append_user_classpath(option, opt, v
 # Deprecated
 def clean_system_classpath(option, opt, value, parser):
     deprecation_notice()
-    if os.getuid() is 0:
+    if os.getuid() == 0:
         manager.clean_classpath(system_classpath_target())
         update_env()
     else:
