From 2489106dc048fde4b1cf65b242a7901f57d6f923 Mon Sep 17 00:00:00 2001
From: Darrion Whitfield <dwhitfield@ramfield.net>
Date: Sun, 10 Aug 2025 14:17:47 -0500
Subject: [PATCH] _: adding the right website to the build

---
 build.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build.rs b/build.rs
index efe50a4..a50f5c8 100644
--- a/build.rs
+++ b/build.rs
@@ -27,7 +27,7 @@ fn main() -> io::Result<()> {
 
     let usage   = include_str!("src/usage.txt");
     let tagline = "doge \\1;32m\\0m command-line DNS client";
-    let url     = "https://dns.lookup.dog/";
+    let url     = "https://dog.ramfield.net/";
 
     let ver =
         if is_debug_build() {
