Index: deluge/core/torrent.py
===================================================================
--- a/deluge/core/torrent.py	(revision 25150f13afa82cec11214f26e159d74ec5a4258e)
+++ b/deluge/core/torrent.py	(revision cc69c9c85b7e1ff61c0c4cd5b21b299ba6e00b8c)
@@ -335,5 +335,5 @@
                     break
 
-        self.options["file_priorities"] = handle_file_priorities
+        self.options["file_priorities"] = self.handle.file_priorities()
         if self.options["file_priorities"] != list(file_priorities):
             log.warning("File priorities were not set for this torrent")
