--- deluge/core/alertmanager.py~	2013-02-25 18:01:07.000000000 +0100
+++ deluge/core/alertmanager.py	2013-03-01 14:34:51.356633948 +0100
@@ -51,7 +51,7 @@
 class AlertManager(component.Component):
     def __init__(self):
         log.debug("AlertManager initialized..")
-        component.Component.__init__(self, "AlertManager", interval=0.3)
+        component.Component.__init__(self, "AlertManager", interval=1.0)
         self.session = component.get("Core").session
 
         self.session.set_alert_mask(
