diff -rup a/deluge/core/alertmanager.py b/deluge/core/alertmanager.py
--- a/deluge/core/alertmanager.py	2019-06-08 22:57:25.000000000 +0200
+++ b/deluge/core/alertmanager.py	2019-10-13 17:01:44.720306086 +0200
@@ -42,7 +42,7 @@ class AlertManager(component.Component):
 
     def __init__(self):
         log.debug('AlertManager init...')
-        component.Component.__init__(self, 'AlertManager', interval=0.3)
+        component.Component.__init__(self, 'AlertManager', interval=1.0)
         self.session = component.get('Core').session
 
         # Increase the alert queue size so that alerts don't get lost.
