diff --git a/pango/fonts.c b/pango/fonts.c
index d9a07c0..8cdb169 100644
--- a/pango/fonts.c
+++ b/pango/fonts.c
@@ -2261,3 +2261,13 @@ pango_font_face_list_sizes (PangoFontFace  *face,
       *n_sizes = 0;
     }
 }
+
+/* Compatibility hacks */
+
+int
+pango_font_metrics_get_height (PangoFontMetrics *metrics)
+{
+  g_return_val_if_fail (metrics != NULL, 0);
+    return (pango_font_metrics_get_ascent(metrics)+
+            pango_font_metrics_get_descent(metrics))/PANGO_SCALE;
+}
diff --git a/pango/pango-font.h b/pango/pango-font.h
index 4af31a9..af2c2ff 100644
--- a/pango/pango-font.h
+++ b/pango/pango-font.h
@@ -690,6 +690,9 @@ struct _PangoFontClass
 #define PANGO_GLYPH_UNKNOWN_FLAG    ((PangoGlyph)0x10000000)
 #define PANGO_GET_UNKNOWN_GLYPH(wc) ((PangoGlyph)(wc)|PANGO_GLYPH_UNKNOWN_FLAG)
 
+/* Compatibility hacks */
+PANGO_AVAILABLE_IN_1_42
+int pango_font_metrics_get_height (PangoFontMetrics *metrics);
 
 G_END_DECLS
 
