--- a/epm	2013-12-16 19:33:23.102231517 +0100
+++ b/epm.new	2013-12-16 19:35:03.123854072 +0100
@@ -191,15 +191,18 @@
     my %vals = map {$_, '(n/a)'} qw/CATEGORY HOSTNAME DESCRIPTION URL LICENSE NAME RELEASE VERSION SLOT PACKAGER/;
     my $ldbpath = "$dbpath/$p";
     my $fmt = <<EOT;
-Name        : \%s
-Version     : \%s
-Release     : \%-28s        Slot: \%s
-Install date: \%-28s  Build Host: \%s
-Group       : \%-28s     License: \%s
-Size        : \%s
-Packager    : \%s
-URL         : \%s
-Summary     : \%s
+Name         : \%s
+Version      : \%s
+Release      : \%-28s
+Slot         : \%s
+Install date : \%-28s
+Build Host   : \%s
+Group        : \%-28s
+License      : \%s
+Size         : \%s
+Packager     : \%s
+URL          : \%s
+Summary      : \%s
 EOT
     # Extract some information from files stored in the Portage pkgdb
     open(F, "bzcat $ldbpath/environment.bz2|")
