diff -rup sysdig-0.17.0/userspace/libsinsp/marathon_http.cpp sysdig-0.17.0-fix/userspace/libsinsp/marathon_http.cpp
--- sysdig-0.17.0/userspace/libsinsp/marathon_http.cpp	2017-07-24 18:13:14.000000000 +0200
+++ sysdig-0.17.0-fix/userspace/libsinsp/marathon_http.cpp	2017-08-10 15:01:48.734401583 +0200
@@ -7,7 +7,6 @@
 #include "marathon_http.h"
 #include "curl/curl.h"
 #include "curl/easy.h"
-#include "curl/curlbuild.h"
 #define BUFFERSIZE 512 // b64 needs this macro
 #include "b64/encode.h"
 #include "sinsp.h"
diff -rup sysdig-0.17.0/userspace/libsinsp/mesos_http.cpp sysdig-0.17.0-fix/userspace/libsinsp/mesos_http.cpp
--- sysdig-0.17.0/userspace/libsinsp/mesos_http.cpp	2017-07-24 18:13:14.000000000 +0200
+++ sysdig-0.17.0-fix/userspace/libsinsp/mesos_http.cpp	2017-08-10 15:01:36.934302944 +0200
@@ -7,7 +7,6 @@
 #include "mesos_http.h"
 #include "curl/curl.h"
 #include "curl/easy.h"
-#include "curl/curlbuild.h"
 #include "sinsp.h"
 #include "sinsp_int.h"
 #include "sinsp_curl.h"
diff -rup sysdig-0.17.0/userspace/sysdig.project sysdig-0.17.0-fix/userspace/sysdig.project
--- sysdig-0.17.0/userspace/sysdig.project	2017-07-24 18:13:14.000000000 +0200
+++ sysdig-0.17.0-fix/userspace/sysdig.project	2017-08-10 15:01:22.184181224 +0200
@@ -752,7 +752,6 @@
             </VirtualDirectory>
             <VirtualDirectory Name="include">
               <VirtualDirectory Name="curl">
-                <File Name="../build/release/curl-prefix/src/curl/include/curl/curlbuild.h"/>
                 <File Name="../build/release/curl-prefix/src/curl/include/curl/easy.h"/>
                 <File Name="../build/release/curl-prefix/src/curl/include/curl/stdcheaders.h"/>
                 <File Name="../build/release/curl-prefix/src/curl/include/curl/curlrules.h"/>
