From b7d0830ea10f7c27d7369aaedf2f480824dc426f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Holger=20Hoffst=C3=A4tte?= <holger.hoffstaette@googlemail.com>
Date: Thu, 4 Jun 2015 19:02:14 +0200
Subject: [PATCH] Use tsk_seruntime() to collect runtime info instead of scheduler-specific struct access.

---
 driver/ppm_cputime.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/driver/ppm_cputime.c b/driver/ppm_cputime.c
index 9e82e24..caf0164 100644
--- a/driver/ppm_cputime.c
+++ b/driver/ppm_cputime.c
@@ -239,7 +239,11 @@ static void cputime_adjust(struct task_cputime *curr,
 void ppm_task_cputime_adjusted(struct task_struct *p, cputime_t *ut, cputime_t *st)
 {
 	struct task_cputime cputime = {
-		.sum_exec_runtime = p->se.sum_exec_runtime,
+#ifdef CONFIG_SCHED_BFS
+ 		.sum_exec_runtime = tsk_seruntime(p),
+#else
+		.sum_exec_runtime = p->se.sum_exec_runtime,
+#endif
 	};
 
 	task_cputime(p, &cputime.utime, &cputime.stime);
