
--- openjdk/hotspot/src/share/vm/opto/loopnode.cpp	2020-01-21 11:18:23.023310816 +0100
+++ openjdk/hotspot/src/share/vm/opto/loopnode.cpp	2020-01-21 11:18:21.852301732 +0100
@@ -3548,6 +3548,9 @@
     case Op_AryEq:
       pinned = false;
     }
+    if (UseShenandoahGC && n->is_CMove()) {
+      pinned = false;
+    }
     if( pinned ) {
       IdealLoopTree *chosen_loop = get_loop(n->is_CFG() ? n : get_ctrl(n));
       if( !chosen_loop->_child )       // Inner loop?
