diff -rup icedtea-3.5.0.dist/Makefile.in icedtea-3.5.0/Makefile.in
--- icedtea-3.5.0.dist/Makefile.in	2017-07-21 04:27:47.991098244 +0200
+++ icedtea-3.5.0/Makefile.in	2017-07-22 19:12:21.989218502 +0200
@@ -119,7 +119,9 @@ target_triplet = @target@
 @WITH_ALT_HSBUILD_TRUE@	patches/hotspot/$(HSBUILD)/8179084-pr3409-rh1455694.patch \
 @WITH_ALT_HSBUILD_TRUE@	patches/hotspot/$(HSBUILD)/8180048-pr3411-rh1449870.patch \
 @WITH_ALT_HSBUILD_TRUE@	patches/hotspot/$(HSBUILD)/8181055-pr3394-rh1448880.patch \
-@WITH_ALT_HSBUILD_TRUE@	patches/hotspot/$(HSBUILD)/8181946.patch
+@WITH_ALT_HSBUILD_TRUE@	patches/hotspot/$(HSBUILD)/8181946.patch \
+@WITH_ALT_HSBUILD_TRUE@	patches/hotspot/$(HSBUILD)/icedtea-3.5.0-shenandoahCollectorPolicy.cpp.patch \
+@WITH_ALT_HSBUILD_TRUE@	patches/hotspot/$(HSBUILD)/icedtea-3.5.0-shenandoahTaskqueue.hpp.patch
 
 
 # AArch32 doesn't yet have the latest update
