--- Makefile.in~	2020-01-21 02:50:51.000000000 +0100
+++ Makefile.in	2020-01-22 01:16:09.598574207 +0100
@@ -132,6 +132,7 @@ target_triplet = @target@
 # Shenandoah has the AArch64 port so should get the return value & debug build fixes
 # Shenandoah also needs some fixes to build on all architectures
 @WITH_ALT_HSBUILD_TRUE@@WITH_SHENANDOAH_HSBUILD_TRUE@am__append_4 = \
+@WITH_ALT_HSBUILD_TRUE@@WITH_SHENANDOAH_HSBUILD_TRUE@	patches/hotspot/$(HSBUILD)/8237007-assert-failure-during-C2-compilation.patch \
 @WITH_ALT_HSBUILD_TRUE@@WITH_SHENANDOAH_HSBUILD_TRUE@	patches/hotspot/$(HSBUILD)/8158260-pr2991-rh1341258-aarch64.patch \
 @WITH_ALT_HSBUILD_TRUE@@WITH_SHENANDOAH_HSBUILD_TRUE@	patches/hotspot/$(HSBUILD)/8222286-pr3727-s390-shenandoah.patch \
 @WITH_ALT_HSBUILD_TRUE@@WITH_SHENANDOAH_HSBUILD_TRUE@	patches/hotspot/$(HSBUILD)/pr3519-return_value-02.patch
